### ISSUE TITLE
Null on Mutation

### ISSUE THREAD

**Issue by netwrkx** [zino-app/graphql-flutter#253]
**Describe the bug**
mutation return null in flutter but works in graphql browser

Response from flutter
I/flutter ( 4727): RESULT
I/flutter ( 4727): Instance of 'QueryResult'
I/flutter ( 4727): null
I/flutter ( 4727): null

**System Setup**
```
[√] Flutter (Channel beta, v1.4.9-hotfix.1, on Microsoft Windows [Version 10.0.17763.404], locale en-US)
    • Flutter version 1.4.9-hotfix.1 at C:\flutter
    • Framework revision 88fa7ea403 (3 weeks ago), 2019-04-11 14:01:46 -0700
    • Engine revision cc115500b9
    • Dart version 2.2.1 (build 2.2.1-dev.3.0 None)

graphql_flutter: 1.0.0+4
```

**To Reproduce**
```
mutation sample
mutation {createBusiness(name: "SSSSSS", address: "kkkk", description: "ppppp", phone: "00000",  categoryIds:"", userId:"5cb3853c8456426f40396177"){
  id
  name
}}
```

```
//mutation.createBusiness file
const String createBusiness = r'''
  mutation createBusiness($name: String, $address: String, $description: String, $phone: String, $categoryIds: String) {
    createBusiness(name: $name, address: $address, description: $description, phone: $phone, categoryIds: $categoryIds) {
      id
      name
    }
  }
''';
```

  ```
import 'package:Kball/graphql/mutations/createBusiness.dart' as mutation;
static final HttpLink _httpLink = HttpLink(
    uri: kAppUrl + '/graphql',
  );

  static final AuthLink _authLink = AuthLink(
    getToken: () =>  'Bearer $YOUR_PERSONAL_ACCESS_TOKEN',
  );

  static final Link _link = _authLink.concat(_httpLink as Link);

  static final GraphQLClient _client = GraphQLClient(
    cache: NormalizedInMemoryCache(
      dataIdFromObject: typenameDataIdFromObject,
    ),
    link: _link,
  );

  Future<QueryResult> create(String name, String address, String description, String phone, String category, String userId) async {
    SharedPreferences prefs = await SharedPreferences.getInstance();
    YOUR_PERSONAL_ACCESS_TOKEN = prefs.getString(kKeychainSecret);
     
    final MutationOptions _options = MutationOptions(
        document: mutation.createBusiness, //
        variables: {
          'name': name,
          'address': address,
          'description': description,
          'phone': phone,
          'categoryIds': category,
          'userId': userId
        },
//      fetchPolicy: widget.options.fetchPolicy,
//      errorPolicy: widget.options.errorPolicy,
    );

    final QueryResult result = await _client.mutate(_options);



    print('RESULT');
    print(result);
    print(result.data);
    print(result.data.toString());

    return result;
  }


```

**Comment by eusdima** [comment]
@netwrkx can you use the bug report template? Right now its hard to follow and figure out what the issue is.

**Comment by netwrkx** [comment]
@eusdima i hope its better this way

**Comment by endigo** [comment]
@netwrkx
``` dart
if (result.hasErrors){
print(result.errors.toString());
}
```

**Comment by netwrkx** [comment]
thanks....

```
how to i use input type
i got an error but i think it related to my mutation file
I/flutter ( 5620): [The categoryIds definition is not executable.: Undefined location]
```

```
how to i use input type?
const String createBusiness = r'''
  input categoryIds {
      _id: String
  }

  mutation createBusiness($name: String, $address: String, $description: String, $phone: String, $categoryIds: [categoryIds]) {
    createBusiness(name: $name, address: $address, description: $description, phone: $phone, categoryIds: $categoryIds) {
      id
      name
    }
  }
''';
```

**Comment by endigo** [comment]
You can't write input type on your client side.

But you can pass through variables: 
```dart
final options = MutationOptions(
  document: createBusiness,
  variables: {
   'name': 'Test',
   ...
   'categoryIds': <Map<String, dynamic>>[{
      '_id': '1234567'
   }, {
      '_id': '1234568'
    }]
  }
)
```

**Comment by netwrkx** [comment]
@endigo  Then what data type will categoryIds be in my mutation file.
If name is a String, categoryIds is ??

**Comment by endigo** [comment]
What is type in your GraphQL API?

**Comment by netwrkx** [comment]
Feathersjs with graphql
If name is a String, categoryIds is ?? In flutter mutation file

**Comment by netwrkx** [comment]
@endigo how do I represent array of string or Json list type in my mutation file since I can't use input type on client-side

**Comment by endigo** [comment]
input types are backend side feature, so you can use json in categoryids

**Comment by netwrkx** [comment]
Thanks
